### PR TITLE
gh/workflows: stop using ubuntu-18.04 runner

### PR DESCRIPTION
**Pull request by julianwiedmann**
The ubuntu-18.04 runner is deprecated, and GH has introduced brownouts so
that users start to migrate. Move on to ubuntu-20.04.

See https://github.com/actions/runner-images/issues/6002

Signed-off-by: Julian Wiedmann <jwi@isovalent.com>

### CODE REVIEW

File: .github/workflows/documentation.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ concurrency:
 jobs:
   check_changes:
     name: Deduce required tests from code changes
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
```
Comment (sayboras) [inline .github/workflows/documentation.yaml:20]: should we go to 22.04 directly ? :thinking:

Comment (julianwiedmann) [reply .github/workflows/documentation.yaml:20]: No clue :man_shrugging:

**Review by sayboras — APPROVED** [review]
:shipit:

File: .github/workflows/documentation.yaml (hunk 1 above)
Comment (tklauser) [reply .github/workflows/documentation.yaml:20]: `ubuntu-latest` is currently still [pointing](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) to `ubuntu-20.04`, so I think 20.04 is fine for now.

**Review by tklauser — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Pull request merged**